core/src/sql/statements/set.rs

```diff
@@ -35,7 +37,28 @@ impl SetStatement {
 		// Check if the variable is a protected variable
 		match PROTECTED_PARAM_NAMES.contains(&self.name.as_str()) {
 			// The variable isn't protected and can be stored
-			false => self.what.compute(stk, ctx, opt, doc).await,
+			false => {
+				let result = self.what.compute(stk, ctx, opt, doc).await?;
+				match self.kind {
+					Some(ref kind) => match result.coerce_to(kind) {
+						Ok(r) => Ok(r),
+						Err(Error::CoerceTo {
+							from,
+							into,
+						}) => Err(Error::SetCheck {
+							value: from.to_string(),
+							name: self.name.to_string(),
+							check: into,
+						}),
+						Err(_) => Err(Error::SetCheck {
+							value: "internal error".to_string(),
+							name: format!("${}", self.name),
+							check: kind.to_string(),
+						}),
```

Comment:
I don't think we should cover up errors outside rewriting `CoerceTo` to `SetCheck`, they could provide important context

Reply:
I agree, but can coerceto return anything other than a coerce error?